src/lib/storage/chunk.cpp

```diff
@@ -14,27 +14,45 @@
 
 namespace opossum {
 
-void Chunk::add_segment(std::shared_ptr<BaseSegment> segment) {
-  // Implementation goes here
-}
+void Chunk::add_segment(std::shared_ptr<BaseSegment> segment) { segments.push_back(segment); }
 
 void Chunk::append(const std::vector<AllTypeVariant>& values) {
-  // Implementation goes here
-}
+  DebugAssert(!values.empty(), "");
+  DebugAssert(values.size() == segments.size(), "");
 
-std::shared_ptr<BaseSegment> Chunk::get_segment(ColumnID column_id) const {
-  // Implementation goes here
-  return nullptr;
+  int size = segments.size();
+  for (int i = 0; i < size; i++) {
+    segments[i]->append(values[i]);
+  }
 }
 
+std::shared_ptr<BaseSegment> Chunk::get_segment(ColumnID column_id) const { return segments[column_id]; }
```

Comment:
```suggestion
std::shared_ptr<BaseSegment> Chunk::get_segment(ColumnID column_id) const { return segments.at(column_id); }
```
`at()` will throw an error when accessing indexes out of range